crates/bevy_ecs/src/query/mod.rs

```diff
@@ -57,6 +57,102 @@ mod tests {
 
     #[test]
     fn query_filtered_exactsizeiterator_len() {
+        fn assert_all_sizes_iterator_equal(
+            iterator: impl ExactSizeIterator,
+            expected_size: usize,
+            query_type: &'static str,
+        ) {
+            let len = iterator.len();
+            let size_hint_0 = iterator.size_hint().0;
+            let size_hint_1 = iterator.size_hint().1;
+            // `count` tests that not only it is the expected value, but also
+            // the value is accurate to what the query returns.
+            let count = iterator.count();
+            // This will show up when one of the asserts in this function fails
+            println!(
+                r#"query declared sizes:
+for query:     {query_type}
+expected:      {expected_size}
+len:           {len}
+size_hint().0: {size_hint_0}
+size_hint().1: {size_hint_1:?}
+count():       {count}"#
```

Comment:
```suggestion
                "query declared sizes: \
                for query:     {query_type} \
                expected:      {expected_size} \
                len:           {len} \
                size_hint().0: {size_hint_0} \
                size_hint().1: {size_hint_1:?} \
                count():       {count}"
```
Nit, non blocking, but formatting can be improved